### PR TITLE
sync URL param and store

### PR DESCRIPTION
**Pull request by zhouzilong2020**
The params in the URL will be in the form of: `/?cloudProvider=AWS&region=US+East+(N.+Virginia),US+West+(Oregon)&engineType=MYSQL&chargeType=OnDemand&minCPU=0&minRAM=0&keyword=`

### CODE REVIEW

**Comment by zhouzilong2020** [comment]
PTAL

**Review by LiuJi-Jim — APPROVED** [review]
LGTM

File: frontend/src/layout/AppLayout.vue [hunk 1]
```diff
@@ -93,13 +94,25 @@ const handleUpdateMinCPU = (val: any) => {
...
-  () => [searchConfigStore.searchConfig],
+  () => searchConfigStore.searchConfig,
   () => {
     const config = searchConfigStore.searchConfig;
-    router.replace({
+    const queryParam: RouteParam = {
+      cloudProvider: config.cloudProvider,
+      region: config.region.join("-"),
+      engineType: config.engineType.join("-"),
+      chargeType: config.chargeType.join("-"),
```
Comment (LiuJi-Jim) [inline frontend/src/layout/AppLayout.vue:104]: This brings a limitation that `region`, `engineType`, `chargeType` cannot contain the character `"-"`, please watch out.

Comment (d-bytebase) [reply frontend/src/layout/AppLayout.vue:104]: Regions can contain dash, e.g. us-central1 https://cloud.google.com/compute/docs/regions-zones

File: frontend/src/router/index.ts [hunk 2]
```diff
@@ -33,6 +34,26 @@ export const router = createRouter({
...
+  }
+
+  // typed in URL with param
+  if (!from.name) {
+    const config = useSearchConfigStore().searchConfig;
+    const query = to.query;
+
+    config.cloudProvider = query.cloudProvider as CloudProvider;
+    config.region = (query.region as string).split("-");
+    config.chargeType = (query.chargeType as string).split("-") as ChargeType[];
```
Comment (d-bytebase) [inline frontend/src/router/index.ts:50]: Is it natural to use commas?

**Pull request merged**